src/reanimated2/utils.ts

```diff
@@ -32,3 +32,8 @@ export function isSharedValue<T>(value: any): value is SharedValue<T> {
   'worklet';
   return value?._isReanimatedSharedValue === true;
 }
+
+export function isWorklet(value: unknown) {
+  // TODO: mark value as worklet
+  return typeof value === 'function' && '__workletHash' in value;
```

Comment:
this implementation changes the behavior. We should only test for the value of `__workletHash` here, I think that if you take any HostFunction it will pass this check here because it is a function and `in` doesn't work properly for JSI objects and always return true

Reply:
After dealing with HostObjects I agree that `value.__workletHash` should be the only check here.

Reply:
@tjzel Another benefit of `isWorklet` function would be that we could assume that the provided value is of a worklet type. Feel free to perform a friendly take-over of this PR or close it.